src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/GeneratedCodeContainer.cs

```diff
@@ -39,34 +95,59 @@ public TService GetService<TService>()
             return default(TService);
         }
 
-        public void SetOutput(SourceText source, RazorCodeDocument codeDocument)
+        public void SetOutput(DocumentSnapshot document)
         {
-            Source = source;
-            Output = codeDocument.GetCSharpDocument();
+            lock (_setOutputLock)
+            {
+                if (!document.TryGetTextVersion(out var version))
+                {
+                    return;
+                }
```

Comment:
In what cases can this happen? When the document hasn't been read yet? Should this crash?

Reply:
No case we support today. It's a `TryGet` so figured I'd just be protective here. We could explode to let the user know they're doing something unexpected.

Reply:
user?

Reply:
lol sure, user being our future selves 😉

Reply:
Only one place should be calling this code, that's my point.